src/js/angular/ttyg/directives/chat-panel.directive.js

```diff
@@ -0,0 +1,180 @@
+ * regenerating chat content, and other interactive features.
+ *
+ * It encapsulates the logic and presentation for displaying and managing chat-related interactions.
+ *
+ * @example
+ * <chat-panel></chat-panel>
+ */
+function ChatPanelComponent(toastr, $translate, TTYGContextService, TTYGService) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/angular/ttyg/templates/chat-panel.html',
+        scope: {},
+        link: ($scope, element, attrs) => {
+
+            // =========================
+            // Public variables
+            // =========================
+
+            /**
+             * @type {ChatModel}
+             */
+            $scope.chat = undefined;
+
+            /**
+             * @type {AgentModel}
+             */
+            $scope.selectedAgent = undefined;
+
+            /**
+             * @type {ChatQuestion}
+             */
+            $scope.chatQuestion = undefined;
+
+            // =========================
+            // Private variables
+            // =========================
+
+            // =========================
+            // Public functions
+            // =========================
+
+            /**
+             * Handles the ask question action.
+             */
+            $scope.ask = () => {
+                if (!$scope.chatQuestion.conversationId) {
+                    createNewChat();
+                } else {
+                    askQuestion();
+                }
+            };
+
+            // =========================
+            // Private functions
+            // =========================
+
+            const setupNewChatQuestion = () => {
+                $scope.chatQuestion = getEmptyChatQuestion();
+            };
+
+            const createNewChat = () => {
+                TTYGContextService.emit(TTYGEventName.CREATE_CHAT, $scope.chatQuestion);
+            };
+
+            const askQuestion = () => {
+                if ($scope.chat) {
+                    $scope.chat.appendMessage(chatQuestionToChatMessageMapper($scope.chatQuestion));
+                }
+                const question = cloneDeep($scope.chatQuestion);
+                setupNewChatQuestion();
+                TTYGContextService.emit(TTYGEventName.ASK_QUESTION, question);
+            };
+
+            /**
+             * Handles answering a chat question.
+             *
+             * @param {ChatMessageModel} answer
+             */
+            const onQuestionAnswer = (answer) => {
+                if ($scope.chat && $scope.chat.id === answer.conversationId) {
+                    $scope.chat.appendMessage(answer);
+                }
+            };
+
+            /**
+             * Handles the change of the selected chat.
+             * @param {ChatModel} chat - the new selected chat.
+             */
+            const onChatChanged = (chat) => {
+                if (!chat) {
```

Comment:
I'm a bit confused for this. How a chat can be changed and to not have a chat here?

Reply:
This happens when the 'Create Chat' button is clicked, but the chat has not been created yet. In this case, we will send a request without a chat ID, and the backend will create it.

Reply:
Added comment